projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx

```diff
@@ -46,14 +52,44 @@ export default function useProductCheckoutWorkflow( {
 		from,
 	} );
 
-	// Build the checkout URL.
-	const checkoutProductUrl = getProductCheckoutUrl(
-		productSlug,
+	const checkoutUrl = useMemo( () => {
+		const origin = getCalypsoOrigin();
+		const useConnetAfterCheckoutFlow = ! isRegistered && connectAfterCheckout;
+
+		const checkoutPath = useConnetAfterCheckoutFlow
+			? 'checkout/jetpack/'
+			: `checkout/${ siteSuffix }/`;
+
+		const productCheckoutUrl = new URL( `${ origin }${ checkoutPath }${ productSlug }` );
+
+		if ( useConnetAfterCheckoutFlow ) {
+			productCheckoutUrl.searchParams.set( 'connect_after_checkout', true );
+			productCheckoutUrl.searchParams.set( 'admin_url', adminUrl );
+			productCheckoutUrl.searchParams.set( 'from_site_slug', siteSuffix );
```

Comment:
@sirbrillig mentioned this in the corresponding Calypso PR - I wonder if we can find a more descriptive name for this parameter 🤔 `site_to_connect` maybe?

Reply:
Yes, I agree `from_site_slug` is a bit confusing what it is...  Let me explain:
 `fromSiteSlug` is the Jetpack "site-slug passed _**from the site**_ via url query arg" (into Calypso), for use cases when we want to know the site but there is not a site in context, such as in siteless checkout. As opposed to `siteSlug` which is the site-slug present when the site is in context (ie- the site is available in State, such as when the site is connected and the user is logged in).
My only concern with changing the variable name to `site_to_connect` is that it restricts it's use case to only connecting the site, when the variable could also potentially be used for other purposes in which we need to know the site but we cannot get the selected site from State for whatever reason (site not connected, user not logged-in, etc).
Addressing @sirbrilling's feedback in the corresponding Calypso PR, I added comments in all the places the variable is present in Calypso, clearly describing what the variable is and it's purpose, so for now I will do the same thing here and add a comment, clearly describing what the variable/arg is and what it's used for.
That being said, if you still feel strongly about changing the variable to `site_to_connect`, I'm not completely opposed to it at all so please let me know your thoughts and I can change it here and all throughout the Calypso PR.

Reply:
Aha! thanks for that extra context! That makes sense and `site_to_connect` does not seem to be the best fit.

I think adding comments should help clarify 👍